Tests/FluentAssertions.Specs/Equivalency/BasicEquivalencySpecs.cs

```diff
@@ -3515,6 +3524,138 @@ public void When_a_tuple_is_compared_it_should_compare_its_components()
+        {
+            var actual = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "bar", "zip", "foo" }
+            };
+
+            var expected = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "foo", "bar", "zip" }
+            };
+
+            Action act = () => actual.Should().BeEquivalentTo(expected, o => o
+                .ComparingByValue<MyRecord>());
+
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected*of type MyRecord*but found*MyRecord*")
+                .WithMessage("*Compare*MyRecord by value*");
+        }
+
+        [Fact]
+        public void When_all_records_should_be_treated_as_value_types_it_should_use_equality_for_comparing()
+        {
+            var actual = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "bar", "zip", "foo" }
+            };
+
+            var expected = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "foo", "bar", "zip" }
+            };
+
+            Action act = () => actual.Should().BeEquivalentTo(expected, o => o
+                .ComparingRecordsByValue());
+
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected*of type MyRecord*but found*MyRecord*")
+                .WithMessage("*Compare records by value*");
+        }
+
+        [Fact]
+        public void When_all_records_except_a_specific_type_should_be_treated_as_value_types_it_should_compare_that_specific_type_by_its_members()
+        {
+            var actual = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "bar", "zip", "foo" }
+            };
+
+            var expected = new MyRecord
+            {
+                StringProperty = "foo",
+                CollectionProperty = new[] { "foo", "bar", "zip" }
+            };
+
+            actual.Should().BeEquivalentTo(expected, o => o
+                .ComparingRecordsByValue()
+                .ComparingByMembers<MyRecord>());
+        }
+
+        [Fact]
+        public void When_global_record_comparing_options_are_chained_it_should_ensure_the_last_one_wins()
+        {
+            var actual = new MyRecord
+            {
+                CollectionProperty = new[] { "bar", "zip", "foo" }
+            };
+
+            var expected = new MyRecord
+            {
+                CollectionProperty = new[] { "foo", "bar", "zip" }
+            };
+
+            actual.Should().BeEquivalentTo(expected, o => o
+                .ComparingRecordsByValue()
+                .ComparingRecordsByMembers());
+        }
+
+        private record MyRecord
+        {
+            public string StringProperty;
```

Comment:
Why is `StringProperty` a Pascal cased field?

Reply:
Because I made it a field later on. Renamed it. But it's still public, so the casing is correct.